packages/react-components/react-utilities/src/compose/resolveShorthand.ts

```diff
@@ -19,22 +19,19 @@ export type ResolveShorthandFunction<Props extends UnknownSlotProps = UnknownSlo
 /**
  * Resolves shorthands into slot props, to ensure normalization of the signature
  * being passed down to getSlots method
- * @param value - the base shorthand props
+ * @param shorthand - the base shorthand props
  * @param options - options to resolve shorthand props
  */
-export const resolveShorthand: ResolveShorthandFunction = (value, options) => {
+export const resolveShorthand: ResolveShorthandFunction = (shorthand, options) => {
   const { required = false, defaultProps } = options || {};
-  if (value === null || (value === undefined && !required)) {
+  if (shorthand === null || (shorthand === undefined && !required)) {
     return undefined;
   }
-
-  let resolvedShorthand = {} as UnknownSlotProps;
-
-  if (typeof value === 'string' || typeof value === 'number' || Array.isArray(value) || isValidElement(value)) {
-    resolvedShorthand.children = value;
-  } else if (typeof value === 'object') {
```

Comment:
I can be wrong, but now the behavior **is different** as `null` went into this condition as `typeof null === 'object'`. I also think that using `isSlot()` there is nice, but might affect performance as we run the same checks again as in `isShorthandValue()` (`isValidElement`, `Array.isArray`)

Reply:
Valid point on the performance, I'll rollback